sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -106,6 +106,12 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     checkKeywordsExist(sql("describe functioN abcadf"), "Function: abcadf not found.")
   }
 
+  test("SPARK-28143: IN expression missing attribute should throw analysis exception") {
+    val query = "select 1 from where in ()"
```

Comment:
BTW, is it correct syntax? Looks like we never supported `in` as function before.

Reply:
It's a parser thing. The parser has no idea whether in is a keyword or function name.

Reply:
`select 1 from where from in ();` 
:-)

Reply:
It is kinda weird, I use this query just because it is simple.

a real world wrong query maybe like this

```sql
select dt, os, userid, usertype
    from tbl1
    where dt=date_sub('2019-06-30',1)
    and source='client'
    and in ('iphone','android')  --- here
```